### PR TITLE
[iOS] Fix issue using Brush transparent color in Frame 

### LINKED ISSUE #13378: [Bug] iOS LinearGradientBrush Transparent Color
### Description
Using this sample xaml:
```
<ContentPage.Content >
		<StackLayout Spacing="0">

			<Frame BorderColor="White"
       HeightRequest="120"
       WidthRequest="120">
				<Frame.Background>
					<LinearGradientBrush EndPoint="0,1">
						<GradientStop Color="Transparent"
                          Offset="0.2" />
						<GradientStop Color="White"
                          Offset="1.0" />
					</LinearGradientBrush>
				</Frame.Background>
			</Frame>
		</StackLayout>

	</ContentPage.Content>
```
shows on iOS a gray area. ;)
### Expected Behavior
All should be white. Like android.

### Actual Behavior
Shows a gray area.

### Basic Information

- Version with issue: 4.8.0.1560
- Last known good version:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library / AndroidX Version: 
- NuGet Packages:
- Affected Devices:

### Environment

<details><summary>Show/Hide Visual Studio info</summary>

```
```

</details>

### Screenshots
iOS:
ios_Brush

Android:
android_Brush

### Workaround
Use skia

### PR DESCRIPTION
**Pull request by jsuarezruiz**
### Description of Change ###

Fix issue using Brush transparent color in Frame on iOS.
The issue is a kind of grey smudge. The reason for the issue is that our gradient is not going from transparent to white. Not at all! It is, in fact, going from  black with zero alpha to white full alpha!. That’s because while our alpha channel is going from 1.0 to 0.0 to achieve the transparency effect, our R, G, and B values are going from 1.0 to 0.0 as well, since the main color of the UIColor.Clear is black.

### Issues Resolved ### 

- fixes #13378

### API Changes ###
 
 None

### Platforms Affected ### 

- iOS

### Behavioral/Visual Changes ###

None

### Before/After Screenshots ### 
fix13378

### Testing Procedure ###
Launch Core Gallery and navigate to the issue 13378. 

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by AlleSchonWeg** [comment]
@jsuarezruiz
Could you merge this into SR 4?

**Comment by jsuarezruiz** [comment]
Team, could we include this requested fix in the upcoming Release?

File: Xamarin.Forms.Platform.iOS/Extensions/BrushExtensions.shared.cs [hunk 1]
```diff
@@ -63,5 +63,35 @@ static NSNumber[] GetCAGradientLayerLocations(List<GradientStop> gradientStops)
...
+				return new CGColor[0];
+
+			CGColor[] colors = new CGColor[gradientStops.Count];
+
+			int index = 0;
+			foreach (var gradientStop in gradientStops)
+			{
+				if (gradientStop.Color == Color.Transparent)
+				{
+					var color = gradientStops[index == 0 ? index + 1 : index - 1].Color;
```
Comment (rachelkang) [inline Xamarin.Forms.Platform.iOS/Extensions/BrushExtensions.shared.cs:79]: Can you help me understand the logic here?

**Review by rachelkang — COMMENTED** [review]
are there any tests you can write for this?

File: Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs [hunk 2]
```diff
@@ -536,7 +536,7 @@ void UpdateBackground()
 						if (backgroundColor.IsDefault)
 							NativeView.BackgroundColor = ColorExtensions.BackgroundColor;
 						else
-							NativeView.BackgroundColor = backgroundColor.ToUIColor();
+							NativeView.BackgroundColor = Element.BackgroundColor.ToUIColor();
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs:539]: What's the reason of this change? wasn't backgroundColor already the color of Element.BackgroundColor ?

Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs:539]: Just reuse the `backgroundColor` variable, but, is the same.

File: Xamarin.Forms.Platform.iOS/Extensions/BrushExtensions.shared.cs (hunk 1 above)
Comment (jsuarezruiz) [reply Xamarin.Forms.Platform.iOS/Extensions/BrushExtensions.shared.cs:79]: If the GradientStop color is transparent, use the next one  Color with Alpha 0. In other case, the effect is not correct because use a Clear default color and can see a black color in the gradient from one to other color. 
This article explain it well: https://betterprogramming.pub/the-proper-way-of-creating-a-transparent-gradient-layer-in-ios-b082daa866b1

File: Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs (hunk 2 above)
Comment (rmarinho) [reply Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs:539]: right but this is another access to the can't we just leave as is ?

**Comment by RenatGaliew** [comment]
Now i am updating Xamarin.Forms to 5.0.0.83

Don't work. 
<Frame.Background>
          <SolidColorBrush Color="#90333232"/>
</Frame.Background>

**Comment by Tommigun1980** [comment]
@RenatGaliew Please file a new bug report if a fix doesn't work, as I am not sure these closed tickets are monitored.
